pkg/backup/resource_backupper.go

```diff
@@ -276,6 +285,80 @@ func (rb *defaultResourceBackupper) backupResource(group *metav1.APIResourceList
 	return nil
 }
 
+// Adds CRD to the backup if one is found corresponding to this resource
+func (rb *defaultResourceBackupper) backupCRD(
+	log logrus.FieldLogger,
+	gr schema.GroupResource,
+	gv schema.GroupVersion,
+	itemBackupper ItemBackupper,
+) {
+	crdGr := schema.GroupResource{Group: apiextv1beta1.GroupName, Resource: "customresourcedefinitions"}
+	crdClient, err := rb.dynamicFactory.ClientForGroupVersionResource(apiextv1beta1.SchemeGroupVersion,
+		metav1.APIResource{
+			Name:       "customresourcedefinitions",
+			Namespaced: false,
+		},
+		"",
+	)
+	if err != nil {
+		log.WithError(err).Error("Error getting dynamic client for CRDs")
+		return
+	}
+
+	unstructuredCRDList, err := crdClient.List(metav1.ListOptions{FieldSelector: fmt.Sprintf("metadata.name=%s", gr.String())})
+	if err != nil {
+		log.WithError(errors.WithStack(err)).Error("Error listing CRDs")
+		return
+	}
+	crds, err := meta.ExtractList(unstructuredCRDList)
+	if err != nil {
+		log.WithError(errors.WithStack(err)).Error("Error extracting CRD list")
+		return
+	}
+	for _, item := range crds {
```

Comment:
I might be misunderstanding the way CRD versioning works, but shouldn't there only be one CRD in this list if found (since two CRDs can't have the same name)? Also, I think we can probably assume that if we're able to fetch a CR, then the version we have is one that is being served by the CRD, so I don't think there's a need to check that the version exists and is being served in the CRD?

Reply:
Ahh, yes I think you're correct here. I was just looking at the versioning model, but since CRDs are cluster-scoped, then there will only be one with a given name. So I can really pull out this whole version-checking bit and assume that if I get a CRD, we back it up. I'll do that.